core/config/config_helper.hpp

```diff
@@ -82,13 +95,35 @@ deferred_factory_parameter<T> get_factory(const pnode& config,
                                           const registry& context,
                                           const type_descriptor& td);
 
-/**
- * specialize for const LinOpFactory
- */
+template <typename T>
+inline deferred_factory_parameter<typename T::Factory> get_specific_factory(
```

Comment:
is this different from `get_factory` followed by a dynamic cast?
If not, this function can be simplified.

Reply:
It has the specific type directly, so it will go the the type::parse without the `type` or `value_type`